interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+        base = os.path.basename(filename)
+        root, _ = os.path.splitext(base)
+        dataset(root)
+    if outName is None and _name != 'noname':
+        outName = _name + '.yaml'
+
+    open_kw = {'encoding': 'latin-1'} if sys.version_info.major == 3 else {}
+    open_mode = 'r' if sys.version_info.major == 3 else 'rU'
+    try:
+        if filename is not None:
+            with open(filename, open_mode, **open_kw) as f:
+                code = compile(f.read(), filename, 'exec')
+        else:
+            code = compile(text, '<string>', 'exec')
+        exec(code)
+    except SyntaxError as err:
+        # Show more context than the default SyntaxError message
+        # to help see problems in multi-line statements
+        if filename:
+            text = open(filename, open_mode).readlines()
+        else:
+            text = text.split('\n')
+        _printerr('%s in "%s" on line %i:\n' % (err.__class__.__name__,
+                                                err.filename,
+                                                err.lineno))
+        _printerr('|  Line |')
+        for i in range(max(err.lineno-6, 0),
+                       min(err.lineno+3, len(text))):
+            _printerr('| % 5i |' % (i+1), text[i].rstrip())
+            if i == err.lineno-1:
+                _printerr(' '* (err.offset+9) + '^')
+        _printerr()
+        sys.exit(3)
+    except Exception as err:
+        import traceback
+
+        if filename:
+            text = open(filename, open_mode).readlines()
+        else:
+            text = text.split('\n')
+            filename = '<string>'
+        tb = traceback.extract_tb(sys.exc_info()[2])
+        lineno = tb[-1][1]
+        if tb[-1][0] == filename:
+            # Error in input file
+            _printerr('%s on line %i of %s:' % (err.__class__.__name__, lineno, filename))
+            _printerr(err)
+            _printerr('\n| Line |')
+
+            for i in range(max(lineno-6, 0),
+                           min(lineno+3, len(text))):
+                if i == lineno-1:
+                    _printerr('> % 4i >' % (i+1), text[i].rstrip())
+                else:
+                    _printerr('| % 4i |' % (i+1), text[i].rstrip())
+        else:
+            # Error in ctml_writer or elsewhere
+            traceback.print_exc()
+
+        sys.exit(4)
+
+    # write the YAML file
+    emitter = yaml.YAML()
+    emitter.width = 70
+
+    for name, cls in globals().items():
+        if hasattr(cls, 'to_yaml'):
+            emitter.register_class(cls)
+
+    with open(outName, 'w') as dest:
+        # information regarding conversion
+        metadata = BlockMap([
+            ('generator', 'cti2yaml'),
+            ('cantera-version', '2.5.0a2'),
+            ('date', formatdate(localtime=True)),
+        ])
+        if filename is not None:
+            metadata['input-files'] = FlowList([base])
+        emitter.dump(metadata, dest)
+
+        outUnits = FlowMap([])
```

Comment:
Would prefer snake case here :smiley:

Reply:
Am I supposed to know what "snake case" is?

Reply:
https://en.wikipedia.org/wiki/Snake_case using underscores, so `out_units` instead of `outUnits`